internal/kafka/test/integration/cloudprovider_test.go

```diff
@@ -360,9 +363,18 @@ func TestListCloudProviderRegions(t *testing.T) {
 				Schedulable:           true,
 			},
 		})
+		dataPlaneClusterConfig = dc
 
 	})
 	defer teardown()
+	var clusterService services.ClusterService
+	h.Env.MustResolve(&clusterService)
+	cluster := dataPlaneClusterConfig.ClusterConfig.GetManualClusters()
+
+	for _, c := range cluster {
+		_, err := common.WaitForClusterStatus(h.DBFactory(), &clusterService, c.ClusterId, api.ClusterReady)
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+	}
```

Comment:
Why is this test modified?

Reply:
Hey @machi1990  After making the changes for this PR These two integration tests started to fail. I believe there was some sort of timing issue that was causing them to fail and @JameelB  helped me fix them.

Reply:
Thanks

Reply:
It's because this change adds additional time to the processMetrics which is the first step in the cluster manager reconcile. This results in the dummy clusters to not be recorded in the database in time for the assertions in this test. 

Having to wait for the cluster to be ready and recorded in the database makes sense here either way since the cloud provider regions endpoint relies on a cluster being there, otherwise it the cloud provider regions endpoint returns an empty list.